packages/shared-ux/chrome/navigation/src/ui/components/navigation_ui.tsx

```diff
@@ -21,17 +21,12 @@ export const NavigationUI: FC<Props> = ({ children, unstyled, footerChildren, da
       {unstyled ? (
         <>{children}</>
       ) : (
-        <EuiFlexGroup
-          direction="column"
-          gutterSize="none"
-          style={{ overflowY: 'auto' }}
-          justifyContent="spaceBetween"
-          data-test-subj={dataTestSubj}
-        >
-          <EuiFlexItem grow={false}>{children}</EuiFlexItem>
-
-          {footerChildren && <EuiFlexItem grow={false}>{footerChildren}</EuiFlexItem>}
-        </EuiFlexGroup>
+        <>
+          <EuiFlyoutBody scrollableTabIndex={-1} data-test-subj={dataTestSubj}>
```

Comment:
I'm surprised to see a flyout component used here? Why did you decide to make this change?

Reply:
To clarify, `EuiCollapsibleNav` (both the old component and the new beta component) have always used `EuiFlyout` under the hood. So using a flyout body and footer as well feels totally natural to me. I also wanted to leave `children` as flexible as possible for usage, but if this is confusing to folks, one option is we could export the the flyout body & footer as `EuiCollapsibleNav.Body` and `EuiCollapsibleNav.Footer`

Reply:
I was just wondering why we can't simply have 2 flex boxes that expand as I put it together in the current version. What does the Flyout component brings?

Reply:
To clarify, the footer **doesn't** expand while the body does (and also flyout bodies automatically adds a scrollbar if content is too tall).

BTW, if you're not using the footer behavior at all, it's totally valid to skip both. The collapsible nav will account for that.

> What does the Flyout component brings?

Some extra styling:

https://github.com/elastic/eui/blob/829709eebc261132af96a154757b6687a31719bd/src/components/collapsible_nav_beta/collapsible_nav_beta.styles.ts#L36-L44

To be clear, you could definitely recreate this yourself if you have to - the CSS isn't terribly complex. But it's just an extra set of small things that get handled out of the box for you by EUI if you do.

Reply:
I followed Cee's examples in the EUI storybook, and it seems to be the right way to go from my point-of-view as well. I'm ++ for the scrollbar behavior

Reply:
Cool, thanks for the explanation 👍